WebAPIBookStore/WebAPIBookStore/Repository/ProductRepository.cs

```diff
@@ -1,49 +1,90 @@
 ﻿using WebAPIBookStore.Data;
 using WebAPIBookStore.Enum;
+using WebAPIBookStore.Input;
 using WebAPIBookStore.Interfaces;
 using WebAPIBookStore.Models;
+using WebAPIBookStore.Result;
 
 namespace WebAPIBookStore.Repository
 {
     public class ProductRepository : IProductRepository
     {
         private readonly DataContext _context;
 
-        public ProductRepository(DataContext context) 
+        public ProductRepository(DataContext context)
         {
             _context = context;
         }
-        public Product? GetProduct(int id)
+        public ProductOutput? GetProductReturnProductOutput(int id)
         {
-            return _context.Products.FirstOrDefault(p => p.Id == id);
+            return _context.Products.Select(p => new ProductOutput
+            {
+                Id = p.Id,
+                Name = p.Name,
+                Description = p.Description,
+                Author = p.Author,
+                Price = p.Price,
+                PublishYear = p.PublishYear,
+                ImageURL = _context.Images.Where(i => i.ProductId == p.Id).Select(i => i.URL).ToList(),
+                CategoryNames = _context.ProductCategories.Where(pc => pc.ProductId == p.Id).Join(_context.Categories, pc => pc.CategoryId, c => c.Id, (pc, c) => c.Name).ToList()
+            }).FirstOrDefault(p => p.Id == id);
```

Comment:
em tính đến performance nhé:
- Bản chất của select là em đang loop qua tùng thằng để nó gán dữ liệu
- Em đang loop hết xong rồi lấy đúng 1 thằng có id thế kia thì khi dữ liệu của em nó to thì em loop cả đống để lấy 1 cái thôi à
=> em firstordefault trước, rồi em mới select nhé